src/Xymat.jl

```diff
@@ -0,0 +1,130 @@
+"""
+    FeTerm{T,S}
+
+Term with an explicit, constant matrix representation
+
+# Fields
```

Comment:
```suggestion
Typically, an `FeTerm` represents the design matrix for the fixed effects. 

!!! note
    `FeTerm` is not the same as [`FeMat`](@ref)!

# Fields
```

I don't know if `FeTerm` is the best name because it doesn't work like a `StatsModels.AbstractTerm`. Maybe make this `FeMat` and `FeMat` into `FeResponseMat` or `XYMat` or `XyMat`?

Reply:
Having now also talked to @kleinschmidt , I think we should change at least one of the names. 

More generally, I'm not sure `FeTerm` and `ReTerm` are ideal, but we're internally consistent in our use of those even though they're not `Term`s in the StatsModel sense, but that latter bit is for a different PR. We could change those names or make them more term like, but again, that's something for a later day.

Reply:
I'm a bit of a purist about the terms `design matrix` and `model matrix`.  To me the term `design matrix` refers to the experimental design, which is to say the levels of the factors in the data table.  The matrix derived from the data table and the formula is the model matrix.  So I would change that to `model matrix`.

Reply:
Easy! I actually originally wrote "model matrix" then remembered that the last time I wrote that, you wanted "design matrix". But now I understand the distinction you're drawing here.